[Paul Lee] iP

paullee18 opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
The **Duke** chat bot helps to keep track of your _todos_! It can store
1. Todo names
2. Deadlines
3. Event timings

**Duke** is 
- Fast
- Simple
- Convenient

If you are curious as to how our Duke program does so many things, here's a snippet of the variables our Duke chat bot stores.

```Java
public class Duke {

    private Storage storage;
    private TaskList taskList;
    private Ui ui;
    private Parser parser;
```

Here are the features: 💯 
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (Coming soon)

Commit:
Add Gradle support

paullee18 committed:
Level 1.

paullee18 committed:
Level-2

paullee18 committed:
Level-3

paullee18 committed:
Level-4

paullee18 committed:
Level-5

paullee18 committed:
Level-6

paullee18 committed:
A-TextUiTesting

paullee18 committed:
Add Javadoc to Duke class

paullee18 committed:
Read and write from duke.txt files so that Task information can be saved and retrieved

paullee18 committed:
Use LocalDate to format dates in Deadline and Event objects

paullee18 committed:
Merge branch 'branch-Level-7'

paullee18 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

paullee18 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

paullee18 committed:
Add OOP to break main code into different classes

paullee18 committed:
Package files into duke package

paullee18 committed:
Add JUnit tests for Parser and TaskList classes

paullee18 committed:
Add dependencies for making JAR file

paullee18 committed:
Add JavaDoc comments to all classes and functions

paullee18 committed:
Add Checkstyle to Gradle and fix Checkstyle violations

paullee18 committed:
Add way for user to find a task by searching for a keyword

paullee18 committed:
Merge branch 'branch-A-JavaDoc'

paullee18 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Task.java

paullee18 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

sugiyem commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,54 @@
+        this.ui.printStart();
+
+        Scanner sc = new Scanner(System.in);
+        boolean exit = false;
```
Maybe a more boolean name for variable exit?

sugiyem commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,103 @@
+        case "deadline":
+            String[] deadlineSplit = fullCommand.split(" /by ");
+            String formattedDate = this.getDate(deadlineSplit[1]);
+            Deadline deadline = new Deadline(deadlineSplit[0].substring(9, deadlineSplit[0].length()), formattedDate);
```
Maybe this line can be broken into multiple lines for easier readability?

sugiyem commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,131 @@
+                String[] lineArr = line.split("#");
+                String className = lineArr[0];
+                String name = lineArr[1];
+                switch(className) {
```
Should have blank space after the switch statement?

sugiyem commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,131 @@
+                    }
+                    break;
+                default:
+
```
Maybe can remove the default case entirely as there is nothing there?

sugiyem commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+     *
+     * @return size of the TaskList
+     */
+    public int size() {
```
Can change the method's name to be more descriptive?

sugiyem commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,103 @@
+        String command = commandBreakdown[0];
+
+        switch (command) {
+        case "list":
```
It's great that you follow the coding standard for case clauses in switch statement.

sugiyem commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,20 @@
+
+public class TaskListTest {
+    @Test
+    public void addingTasks(){
```
Maybe can change the method's name to be a more descriptive test name?

For example, you can rename it to be something like `addTasks_validInput_getCorrectToString`.

sugiyem reviewed (commented):
Generally the code looks good already, but there are some minor coding standard violations as listed below. Also, I think it would be better if you group all the Task classes in one package to make it more organized.

yuehernkang commented:
Overall your code looks good, there seems to be some coding standard violations but overall it is readable.

yuehernkang commented on src/main/java/Duke.java:
```diff
@@ -18,14 +129,15 @@ public static void main(String[] args) {
             String command = sc.nextLine();
             String[] commandBreakdown = command.split(" ");
 
             switch(commandBreakdown[0]) {
```
I think there should be no indentation for the case clauses

paullee18 committed:
Add GUI using Javafx and FXML

paullee18 committed:
Remove unnecessary code from Duke.java due to the fxml files

paullee18 committed:
Refactor code from Parser to make code more readable

paullee18 committed:
Added assert statements to make sure code is working as per assumptions

paullee18 committed:
Merge pull request #2 from paullee18/branch-A-Assertions

Added assert statements to make sure code is working as per assumptions

paullee18 committed:
Merge pull request #1 from paullee18/branch-A-CodeQuality

Refactor code from Parser to make code more readable

paullee18 committed:
Sort tasks whenever the full task list is requested

Tasks are not sorted.

Sorted tasks is more user-friendly, as Events, Deadlines and Todos will be displayed in sequence. Events and Deadlines
with earlier durations and deadlines are also displayed first.

Task implementing Comparable allows us to give an order to the various types of Tasks. For example, Deadlines can come before Events
followed by Todos. Within Deadlines and Events, they can also be sorted by their deadlines and durations.

The TaskList is only sorted when the user gives the list command so that it is more efficient.

paullee18 committed:
Merge branch 'C-sort'

paullee18 committed:
Add more Duke Exceptions to handle more errors I anticipate would be encounterd. Also fix all CheckStyle issues.

paullee18 committed:
Change name of user interface screenshot to Ui.png

paullee18 committed:
Update README.md

paullee18 committed:
Change file structure to include folders for dukeexceptions and gui

paullee18 committed:
Merge branch 'master' of https://github.com/paullee18/ip

paullee18 committed:
Update README.md
